allauth/socialaccount/providers/apple/client.py

```diff
@@ -0,0 +1,88 @@
+import requests
+from time import time
+from urllib.parse import parse_qsl
+
+from django.conf import settings
+from django.utils.http import urlencode
+
+import jwt
+
+from allauth.socialaccount.providers.oauth2.client import (
+    OAuth2Client,
+    OAuth2Error,
+)
+
+
+class Scope(object):
+    EMAIL = 'email'
+    NAME = 'name'
+
+
+class AppleOAuth2Client(OAuth2Client):
+    """
+    Custom client because `Sign In With Apple`:
+        * requires `response_mode` field in redirect_url
+        * requires special `client_secret` as JWT
+    """
+
+    def generate_client_secret(self):
+        """Create a JWT signed with an apple provided private key"""
+        APPLE_PROVIDER_SETTINGS = getattr(
+            settings, 'SOCIALACCOUNT_PROVIDERS', {}
+        ).get('apple', {})
+
+        MEMBER_ID = APPLE_PROVIDER_SETTINGS.get('MEMBER_ID', None)
+        SECRET_KEY = APPLE_PROVIDER_SETTINGS.get('SECRET_KEY', None)
+
+        CURRENT_TIMESTAMP = int(time())
+        claims = {
+            'iss': MEMBER_ID,
+            'aud': 'https://appleid.apple.com',
+            'sub': self.consumer_key,
+            'iat': CURRENT_TIMESTAMP,
```

Comment:
It can  be written as following

```suggestion
            'iat': timezone.now(),
            exp': timezone.now() + timedelta(hours=1),
```

Reply:
Thanks! I am going to update to use `datetime.utcnow()` though, since I don't want to give the impression that timezones matter in the JWT creation.